File: new GP/translator/src/main/resources/MainActivity.java
package com.gradproject.agecalc;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Calendar;

public class MainActivity extends AppCompatActivity {
    EditText dob;
    TextView result;
    Button calc;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        dob = findViewById(R.id.EnterDOB);
        result = findViewById(R.id.textView);
        calc = findViewById(R.id.button1);
    }
    public void getAgeButton(View view){

        int myDOB = Integer.parseInt( dob.getText().toString());
        int currentYear = 2021;
        int age = currentYear - myDOB;
        result.setText("Your age is "+ age +" years");


    }
}

File: new GP/translator/src/main/resources/kt/MainActivity.kt
package com.badawy3.app1

import  android.graphics.drawable.DrawableContainer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button1.setOnClickListener({

            val yearofbirth: Int = EnterDOB.text.toString().toInt()
            val age: Int = 2020 - yearofbirth
            textView.text = "your age is  " + age + " years"

        }
        )


    }

}